src/zenml/zen_server/routers/projects_endpoints.py

```diff
@@ -555,6 +555,49 @@ def create_pipeline(
     return zen_store().create_pipeline(pipeline=pipeline)
 
 
+
+@router.get(
+    "/{project_name_or_id}" + RUNS,
+    response_model=List[PipelineRunResponseModel],
+    responses={401: error_response, 404: error_response, 422: error_response},
+)
+@handle_exceptions
+def list_runs(
```

Comment:
I don't get why this needs to exists, we already have an endpoint that does exactly the same

Reply:
basically we have all list and creates on resources twice, once on the project resource and once on its own. In general now that we are going for project scoping on the dashboard side, all the list queries are explicitly moving to the project scoped endpoints. Thats where we realized that the runs endpoint was missing. This change just makes runs behave like all other resources.

Reply:
instead of calling `/projects/{project_id}/runs` he can just use `/runs?project_name_or_id={project_id}`

Reply:
the issue I see is that we have those functions duplicated now which opens the door to a lot of bugs and inconsistencies IMO

Reply:
theoretically yes, difference being 
`/projects/{project_id}/runs` is basically `project_id: UUID` 
while
`/runs?project_name_or_id={project_id}` equates to  `project_id: Optional[UUID] = None`

As we are slowly transitioning to real role based access control I believe it makes sense to enforce the scope now already to make the transition earlier, then we can let Fastapi do the heavy lifting when the time comes.

the /project endpoints can then be scoped to users who have the right permissions in that particular project

Reply:
i.e., the RestZenStore methods hit a different endpoint than the dashboard now so the CLI and dashboard might show different results for the same request

Reply:
My proposal would be to have duplicates for the time being until projects are firmly established as the scoping mechanism in frontend and clientside, then I would propose to deprecate the duplicates unscoped endpoints